### ISSUE TITLE
Operator Hub: community-operators pod in ImagePullBackoff after cluster update

### ISSUE THREAD

**Issue by jomeier** [openshift/okd#92]
Hi,

yesterday and last time today I updated OKD 4.4 with 

```
oc adm upgrade --to-image=registry.svc.ci.openshift.org/origin/release:4.4.0-0.okd-2020-03-03-052027 --allow-explicit-upgrade --allow-upgrade-with-warnings=true --force=true
```

After the update everything looked fine but as I tried to install something from the Operator Hub through the Web UI the catalog was not there.

The pod **community-operators** in **openshift-marketplace** had an Imagepullbackoff. After I restarted the pod the Operator Hub UI was populated again.

This happend the second time after a cluster upgrade.

release image:
registry.svc.ci.openshift.org/origin/release:4.4.0-0.okd-2020-03-03-052027

Seems like a race condition. Not mission critical for the beta but maybe easy to fix.

Greetings,

Josef

**Comment by vrutkovs** [comment]
we need `oc adm must-gather` archive to find out what's up with this

**Comment by jomeier** [comment]
Yes I know. I was preparing that in parallel but the compressed file has over 700MByte (the cluster is running since a week now).

The error happened again the third time after the third upgrade. I think I have to set up a fresh cluster and will do the update again. Afterwards sending the must-gather tarball to you.

**Comment by adarshaj** [comment]
I had this issue too. It seems that old ReplicaSet stays around even when Deployment has updated, I had to delete Deployment and old ReplicaSet and kill operator pod (which would respawn itself), after that the Deployment created the right ReplicaSet on its own.

**Comment by jomeier** [comment]
Its working on a cluster which is not behind a proxy. I try to setup a new cluster behind our proxy and gather the logs.

**Comment by vrutkovs** [comment]
Is it still happening on recent nightlies / betas?

**Comment by jomeier** [comment]
Hi Vadim,

yes some Operators like Rook and ElasticSearch are still missing. My colleague installed a different catalog from OperatorHub and there Rook and other Operators are included. 

They don't have the label "community" but "custom". 

I'm not sure if it's an official strategy of OKD to not include "custom" operators or not. If Red Hat decided that than it should be documented somehow and we could close this issue.

Greetings,

Josef

**Comment by vrutkovs** [comment]
>They don't have the label "community" but "custom"

That means these are still Red Hat owned and not created by community. You'd need a valid pull secret to use these, so they cannot be included in OKD by default

**Comment by jomeier** [comment]
Because noobaa is available on Operatorhub without a pull secret we thought it should be in the default catalog for OKD. Same is for other Operators on the OperatorHub.

So it may be no issue but a selection criteria from Red Hat what comes into OKDs catalog and what not.

**Comment by vrutkovs** [comment]
Oh I see. The catalog would filter out anything which is not "community", not sure what "custom" is for - and there doesn't seem to be a setting to enable other labels

**Comment by jomeier** [comment]
That’s exactly what we scratch our heads since a few month about.

**Comment by jomeier** [comment]
I had the problem with the operatorhub ui being not available sometimes before the beta with nightlies. I didn’t see it again since then. Because this was the initial problem reported here we can close this one. I opened a different issue for the missing operators in the operatorhub catalog.
